### PR TITLE
CIV-6214 dashboard defendant startus p2

### PR DESCRIPTION
**Pull request by TatsyMaslova**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-6214

### Change description ###

This is a part 2 of defendant dashboard and is raised against the dependant branch that is currently in test. The dashboard statuses that do not currently apply to Lr vs LiP just have place holders. i.e. they will always return false. This is until we develop a corresponding journey. Then we can add logic for the place holder

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/cmc/model/Settlement.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Data
+public class Settlement {
+    private List<PartyStatement> partyStatements;
+
+    @JsonIgnore
+    public boolean isAcceptedByClaimant() {
+        boolean isAcceptedByClaimant = getPartyStatementStream()
```
Comment (PaulH-HMCTS) [inline src/main/java/uk/gov/hmcts/reform/cmc/model/Settlement.java:23]: Move the two boolean conditions into separate methods with more readable names

File: src/test/java/uk/gov/hmcts/reform/cmc/model/SettlementTest.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
+package uk.gov.hmcts.reform.cmc.model;
```
Comment (PaulH-HMCTS) [inline src/test/java/uk/gov/hmcts/reform/cmc/model/SettlementTest.java:1]: Can you check the formatting on this file, doesnt look right

**Review by laarmada — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/cmc/model/Response.java [hunk 3]
```diff
@@ -7,32 +7,68 @@
...
+import java.math.BigDecimal;
+
 @Getter
 @AllArgsConstructor
 @NoArgsConstructor
 @Builder
 public class Response {
 
-    public RespondentResponseType responseType;
+    private RespondentResponseType responseType;
```
[the remark this answers is not in the document]
Comment (TatsyMaslova) [reply src/main/java/uk/gov/hmcts/reform/cmc/model/Response.java:18]: No impact. I think Kamil accidentally added this as public whereas it shoud be private. its only cmc model representation in civil service and will be removed in release2

File: src/main/java/uk/gov/hmcts/reform/cmc/model/Settlement.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        Stream<PartyStatement> partyStatementsStream = getPartyStatementStream();
+        return partyStatementsStream.anyMatch(partyStatement -> partyStatement.isCounterSigned());
+    }
+
+    public boolean isThroughAdmissions() {
+        Stream<PartyStatement> partyStatementsStream = getPartyStatementStream();
+        return partyStatementsStream.filter(PartyStatement::hasOffer)
+            .reduce((first, second) -> second)
+            .stream()
+            .noneMatch(offer -> offer.hasPaymentIntention());
```
Comment (LeonardoPalmeiro) [inline src/main/java/uk/gov/hmcts/reform/cmc/model/Settlement.java:44]: we can replace to    .noneMatch(PartyStatement::hasPaymentIntention);

File: src/main/java/uk/gov/hmcts/reform/cmc/model/Settlement.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    private Stream<PartyStatement> getPartyStatementStream() {
+        Stream<PartyStatement> partyStatementsStream = Optional.ofNullable(partyStatements)
+            .map(Collection::stream).orElseGet(Stream::empty);
+        return partyStatementsStream;
```
Comment (LeonardoPalmeiro) [inline src/main/java/uk/gov/hmcts/reform/cmc/model/Settlement.java:50]: we can replace to
optional.ofNullable(partyStatements.stream()).orElseGet(Stream::empty);

File: src/main/java/uk/gov/hmcts/reform/cmc/model/Settlement.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+            .noneMatch(partyStatement -> partyStatement.isRejected()
+                && partyStatement.isMadeByDefendant());
+        return isAcceptedByClaimant && isNotRejectedByDefendant;
+    }
+
+
+    @JsonIgnore
+    public boolean isSettled() {
+        Stream<PartyStatement> partyStatementsStream = getPartyStatementStream();
+        return partyStatementsStream.anyMatch(partyStatement -> partyStatement.isCounterSigned());
```
Comment (LeonardoPalmeiro) [inline src/main/java/uk/gov/hmcts/reform/cmc/model/Settlement.java:36]: we can use in this way
        return partyStatementsStream.anyMatch(PartyStatement::isCounterSigned);
what do you think?

File: src/main/java/uk/gov/hmcts/reform/cmc/model/Settlement.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    public boolean isThroughAdmissions() {
+        Stream<PartyStatement> partyStatementsStream = getPartyStatementStream();
+        return partyStatementsStream.filter(PartyStatement::hasOffer)
+            .reduce((first, second) -> second)
```
Comment (LeonardoPalmeiro) [inline src/main/java/uk/gov/hmcts/reform/cmc/model/Settlement.java:42]: for me is difficult to understand what is means first, second(can we add more readable name) into .reduce function as well why we need to use .reduce in this case.

File: src/main/java/uk/gov/hmcts/reform/cmc/model/PartyStatement.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+@Builder
+public class PartyStatement {
+
+    private  StatementType type;
+    private  MadeBy madeBy;
+    private  Offer offer;
+
+    @JsonIgnore
+    public boolean isAccepted(){
+       return StatementType.ACCEPTATION == type;
```
Comment (LeonardoPalmeiro) [inline src/main/java/uk/gov/hmcts/reform/cmc/model/PartyStatement.java:21]: I'm not sure if we need to use this.type in this case. 
what do you think?

**Pull request closed without merging**